src/AirshipBindings.NETStandard/AirshipBindings.NETStandard.Android/MessagePageRenderer.cs

```diff
@@ -0,0 +1,166 @@
+            {
+#pragma warning disable CS0618 // Type or member is obsolete
+                base.OnReceivedError(view, errorCode, description, failingUrl);
+#pragma warning restore CS0618 // Type or member is obsolete
+
+                if (renderer.message != null && failingUrl != null && failingUrl.Equals(renderer.message.MessageBodyUrl))
+                {
+                    error = true;
+                }
+            }
+
+            public override void OnClose(WebView webView)
+            {
+                base.OnClose(webView);
+                if (renderer.message != null)
+                {
+                    renderer.messagePage.OnRendererClosed(renderer.message.MessageId);
+                }
+            }
+        }
+
+        private RichPushMessage message;
+        private ICancelable fetchMessageRequest;
+        private UAWebView webView;
+        private MessagePage messagePage;
+        private WebViewClient webViewClient;
+        private string messageId;
+        private FrameLayout nativeView;
+
+        public MessagePageRenderer(Context context) : base(context)
+        {
+            webView = new UAWebView(Context);
+            webViewClient = new MessageWebViewClient(this);
+
+            var fl = new FrameLayout(Context);
+            fl.LayoutParameters = new LayoutParams(LayoutParams.MatchParent, LayoutParams.MatchParent);
+
+            fl.AddView(webView);
+
+            webView.SetWebViewClient(webViewClient);
+            AddView(fl);
+
+            nativeView = fl;
+        }
+
+        public void LoadMessage(string messageId)
+        {
+            if (fetchMessageRequest != null)
+            {
+                fetchMessageRequest.Cancel();
+            }
+
+            message = null;
+            StartLoading(messageId);
+        }
+
+        private void StartLoading(string messageId)
+        {
+            this.message = UAirship.Shared().Inbox.GetMessage(messageId);
+            if (message == null)
+            {
+                fetchMessageRequest = UAirship.Shared().Inbox.FetchMessages(this);
+            }
+            else
+            {
+                if (message.IsExpired)
+                {
+                    messagePage.OnRendererLoadFailed(messageId, false, MessageFailureStatus.Unavailable);
+                    return;
+                }
+            }
+            webView.LoadRichPushMessage(message);
+        }
+
+        protected override void OnElementChanged(ElementChangedEventArgs<Xamarin.Forms.Page> e)
+        {
+            base.OnElementChanged(e);
+
+            if (e.OldElement != null || Element == null)
+            {
+                return;
+            }
+
+            messagePage = e.NewElement as MessagePage;
+            messageId = messagePage.MessageId;
+
+            LoadMessage(messageId);
+        }
+
+        protected override void OnLayout(bool changed, int l, int t, int r, int b)
```

Comment:
This was a non-obvious hack discovered in the forums to get around an apparent Xamarin.Forms bug giving the view the wrong dimensions at runtime